network-ntt/src/server.rs

```diff
@@ -0,0 +1,288 @@
+
+    let (sink, stream) = connection.split();
+    let (sink_tx, sink_rx) = mpsc::unbounded();
+
+    // Processing of the incomming messages.
+    let stream = stream
+        .for_each(move |inbound| {
+            match inbound {
+                Inbound::NothingExciting => future::Either::A(future::ok(())),
+                // New lightweight connection appeared.
+                Inbound::NewConnection(_lwcid) => future::Either::A(future::ok(())),
+                // New node has connected to the server.
+                // We accept the node, we do that immediately without
+                // running future.
+                Inbound::NewNode(lwcid, node_id) => {
+                    sink_tx
+                        .unbounded_send(Message::AckNodeId(lwcid, node_id))
+                        .unwrap();
+                    future::Either::A(future::ok(()))
+                }
+                Inbound::Subscribe(_lwcid, _keep_alive) => {
+                    // TODO: implement subscription mechanism.
+                    //
+                    //state.subscriptions.write().unwrap().insert(
+                    //    SubscriptionId(state.connection.clone(), lwcid),
+                    //    sink_tx.clone(),
+                    //);
+                    future::Either::A(future::ok(()))
+                }
+                Inbound::GetBlockHeaders(lwcid, get_block_header) => {
+                    let sink1 = sink_tx.clone();
+                    let sink2 = sink_tx.clone();
+                    let sink3 = sink_tx.clone();
+                    future::Either::B(future::Either::A({
+                        let mut service = server
+                            .node
+                            .block_service()
+                            .expect("block service is not implemented");
+                        match get_block_header.to {
+                            Some(to) => service.block_headers(&get_block_header.from, &to),
+                            None => service.block_headers_to_tip(&get_block_header.from),
+                        }
+                        .map_err(|err| err.to_string())
+                        .and_then(move |headers| {
+                            Stream::collect(headers)
+                                .map_err(|err| err.to_string())
+                                .and_then(move |hdrs| {
+                                    let msg = Message::BlockHeaders(
+                                        lwcid,
+                                        Response::Ok(BlockHeaders(hdrs)),
+                                    );
+                                    sink1.unbounded_send(msg).map_err(|err| err.to_string())
+                                })
+                        })
+                        .map_err(move |msg| {
+                            let msg = Message::BlockHeaders(lwcid, Response::Err(msg));
+                            sink2.unbounded_send(msg).unwrap_or_else(|_| ());
```

Comment:
... but ignored here and in other `unbounded_send` call sites? I'd rather have an `.unwrap()` everywhere as these unbounded MPSC channel operations shouldn't fail in normal operation.